### PR TITLE
core(server-response-time): fix for lightrider

### PR DESCRIPTION
**Pull request by connorjclark**
This uses the appropriate header in LR to estimate the server response time.

### CODE REVIEW

File: core/audits/server-response-time.js [hunk 1]
```diff
@@ -61,6 +68,14 @@ class ServerResponseTime extends Audit {
     const mainResource = await MainResource.request({devtoolsLog, URL: artifacts.URL}, context);
 
     const responseTime = ServerResponseTime.calculateResponseTime(mainResource);
+    if (responseTime === null) {
+      return {
+        score: 0,
+        runWarnings: [str_(UIStrings.missingTimingWarning)],
+        notApplicable: true,
+      };
+    }
```
Comment (connorjclark) [inline core/audits/server-response-time.js:77]: Maybe we could just throw an error here?

File: core/audits/server-response-time.js [hunk 2]
```diff
@@ -43,10 +45,15 @@ class ServerResponseTime extends Audit {
 
   /**
    * @param {LH.Artifacts.NetworkRequest} record
+   * @return {number|null}
    */
   static calculateResponseTime(record) {
-    const timing = record.timing;
-    return timing ? timing.receiveHeadersStart - timing.sendEnd : 0;
+    // Lightrider does not have timings for sendEnd, but we do have this timing which should be
+    // close.
```
Comment (adamraine) [inline core/audits/server-response-time.js:52]:
```suggestion
    // close to the response time.
```

File: core/audits/server-response-time.js [hunk 3]
```diff
@@ -61,6 +68,14 @@ class ServerResponseTime extends Audit {
     const mainResource = await MainResource.request({devtoolsLog, URL: artifacts.URL}, context);
 
     const responseTime = ServerResponseTime.calculateResponseTime(mainResource);
+    if (responseTime === null) {
```
Comment (adamraine) [inline core/audits/server-response-time.js:71]: Add a test for this case?

File: core/audits/server-response-time.js [hunk 4]
```diff
@@ -43,10 +45,15 @@ class ServerResponseTime extends Audit {
...
+   * @return {number|null}
    */
   static calculateResponseTime(record) {
-    const timing = record.timing;
-    return timing ? timing.receiveHeadersStart - timing.sendEnd : 0;
+    // Lightrider does not have timings for sendEnd, but we do have this timing which should be
+    // close.
+    if (global.isLightrider && record.lrStatistics) return record.lrStatistics.requestMs;
+
+    if (!record.timing) return null;
```
Comment (adamraine) [inline core/audits/server-response-time.js:55]: If we are in LR will `record.timing` always be falsy?

File: core/audits/server-response-time.js (hunk 1 above)
Comment (adamraine) [reply core/audits/server-response-time.js:77]: Yeah throwing an error sounds good unless we are expecting this to happen in a number of cases

File: core/audits/server-response-time.js (hunk 3 above)
Comment (connorjclark) [reply core/audits/server-response-time.js:71]: I'm not aware of a way to force a undefined timing with our unit testing.

File: core/audits/server-response-time.js (hunk 4 above)
Comment (connorjclark) [reply core/audits/server-response-time.js:55]: Not yet, that is to come.

**Review by adamraine — APPROVED** [review]

**Pull request merged**